src/README.md

```diff
@@ -0,0 +1,55 @@
+# Provena Client Architecture Description. 
+
+## Directory Structure Overview: 
+
+- **provenaclient/auth**: This directory contains the interfaces and helper functions supporting the authentication flows used by Provena Python Client of which there a two. A device flow (for manual human auithentication) and an offline flow (for automated or scheduled tasks). This directory also includes a logger configured to track authentication-related requests and responses. Furthermore, helper functions within this directory handle tasks like token storage, expiry, and renewal validation, etc.
+ 
+- **provenaclient/clients**: This directory contains the client interfaces for various API's that the Provena Client interacts with on behalf of the user. It includes separate modules for each Provena microservice such as `auth`, `datastore`, `job-api`, `prov-api`, `search-api`, `registry-api`, and `id-service-api`. Each module contains unique functions that facilitate API calls to the respective hosted API.
+ 
+- **provenaclient/models**: This directory contains Pydantic models that are currently not present in [ProvenaInterfaces](https://pypi.org/project/provena-interfaces/) but are still required within the Client Library. 
+
+- **provenaclient/module**: This directory contains various modules that serve as the user-facing layer of the Provena Python Client. Each module represents an individual API, such as auth, datastore, registry, etc. These modules directly interact with the specific client modules in the provenaclient/clients directory, based on their API mapping (e.g., datastore module -> datastore client), which facilitates API calls to the respective hosted services. Furthermore, within nearly all modules in the provenaclient/modules directory, some submodules are instantiated within the module, managing specific functionalities related to the API (e.g., admin, IO).
+
+- **provenaclient/utils** - This directory contains utility interfaces and helper functions that support the Provena Python Client such as custom exception class, and configuration class. This directory also includes the custom `httpx` client that is encapsulated in an `HttpxClient` class. This class includes custom functions that wrap around the httpx HTTP methods, allowing easy integration and standardisation across the client library.
+
+## Software Architecture and Design:
+
+### Overview 
+The Provena Python Client uses a "layered architecture" that is supported by the Dependency Injection pattern, providing an approach that is modular and allows us to have a separation of concerns. 
+
+For a visual diagram of the layered architecture, refer to this [example of dataset fetch](./fetch-dataset-flow.svg).
```

Comment:
The svg is not a visual diagram of the layered architecture, just a visual diagram of one example functionality that is implemented within and following the architecture